### ISSUE TITLE
Companion app as media_player entity

### ISSUE THREAD

**Issue by bverkron** [home-assistant/android#2512]
**Is your feature request related to a problem? Please describe.**

No

**Describe the solution you'd like**

I would like to be able to "cast" media to the Home Assistant companion app on Android so I can play cloud TTS messages (from Nabu Casa or Google TTS) instead of just the "built-in" TTS that sounds very robotic. This could also serve as a media player for other audio as well.

**Describe alternatives you've considered, if any**

browser_mod which I can't seem to get working, is clunky, and has limitations and awkward workarounds.

**Additional context**

**Comment by dshokouhi** [comment]
Before the companion app can look into this request the `mobile_app` integration needs to first support the `media_player` domain. Please create a feature request for HA core so they can fulfill that before we can proceed here.

**Comment by wired-dev** [comment]
I agree this would be useful.

**Comment by dmiales** [comment]
I confirm, it would be a cool feature. @bverkron, tell me, did you create a task in the core project?

**Comment by bocklucas** [comment]
Agreed this would be sweet, made a feature request on the Home Assistant Forum, https://community.home-assistant.io/t/companion-app-as-media-player/611326

**Comment by pajeronda** [comment]
Good idea!!

**Comment by xteaun** [comment]
Yes! Please implement this! I want to use the companion app with assist in my car while driving!

**Comment by the-mentor** [comment]
+1

**Comment by devbyaccident** [comment]
Would love this! Especially for Music Assistant/Text-to-Speech automations!

**Comment by borgqueenx** [comment]
Please make this...

**Comment by Borsch11** [comment]
YES we need this. Some of the work arounds aka browser mod and FKB are laggy as hell and over bloated. If there was just an addon to add that functionality only it would be awesome but no one seems to have done it.

**Comment by h00g** [comment]
I vote for this since it would fit my need.

I use Android tablets to play music from apps like YouTube Music and the Swedish broadcasting service. They connect to amplifiers via cable or Bluetooth.

The need in my case is for Home Assistant to tell the currently active media app/stream to pause when the power to the amplifier had been shut off.

The companion app exposed what is being played, if it's playing, paused etc but there's no way to tell the device to start/stop playing, which is a very unfortunate and unexpected shortcoming.

In my case battery consumption is not a concern since these tablets are like control panels with 24/7 charging.

If battery use is a topic then a setting enabling/disabling the companion app to expose the device as a media_player may be a consideration.

**Comment by martusi61** [comment]
It would be very useful to me too.

**Comment by HWiese1980** [comment]
Is there no WTH for this request? I would like to vote somewhere! Especially since this request is 3 years old now and still does not seem to be implemented. I see a media_player entity from my companion app, but it does not play media.

**Comment by Gunni** [comment]
> Before the companion app can look into this request the `mobile_app` integration needs to first support the `media_player` domain. Please create a feature request for HA core so they can fulfill that before we can proceed here.

I just made a ticket for core: [Enable `media_player` domain for the `mobile_app` integration](https://github.com/orgs/home-assistant/discussions/3244).
